### PR TITLE
Add test runners for Python 3.10 and update TF and PyTorch versions

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Previously some tests were failing with a modern TF+Python installation.

**Description of the Change:**
Added test runners for Python 3.10. Updated the CI version of TF to 2.10.0 and PyTorch to 1.11.0. 
Also fixed the failing tests.

### CODE REVIEW

**Review by mlxd — APPROVED** [review]
Nice work @eddddddy 
Nothing else to add from me!

**Comment by mlxd** [comment]
Actually, in hindsight, it may also be worth enabling the ` GPU Tests / gpu-tests (pull_request) ` CI runner, since we update both Torch and TensorFlow.

File: .github/workflows/tests-gpu.yml [hunk 1]
```diff
@@ -29,7 +29,6 @@ jobs:
         github.event_name == 'push' ||
         (
           github.event_name == 'pull_request' &&
-          contains(github.event.pull_request.labels.*.name, 'gpu') &&
```
Comment (mlxd) [inline .github/workflows/tests-gpu.yml:32]: We shouldn't remove this. We want to use this as a way to programmatically enable the GPU tests by adding the label "gpu" to the PR.

**Review by mlxd — CHANGES_REQUESTED** [review]

File: .github/workflows/tests-gpu.yml (hunk 1 above)
Comment (mlxd) [reply .github/workflows/tests-gpu.yml:32]: image

After adding the label the GPU tests should run (I just added it now).

Comment (eddddddy) [reply .github/workflows/tests-gpu.yml]: Ah I see 😄

**Review by antalszava — APPROVED** [review]
Looks good! :clap: Thank you :)

**Comment by antalszava** [comment]
1 test seems to be failing: https://github.com/PennyLaneAI/pennylane/runs/8253115898?check_suite_focus=true#step:11:64

**Review by mlxd — APPROVED** [review]
